File: Árbol Binario Interactivo/Arbol.java
import java.util.Scanner;

// Clase para representar un nodo
class Node {
    int data;
    Node left;
    Node right;

    public Node(int data) {
        this.data = data;
        this.left = null;
        this.right = null;
    }
}

class BinaryTreeCreator {
    private static Scanner scanner = new Scanner(System.in);

    // Función para crear un árbol binario de forma recursiva
    public static Node createTree() {
        int choice;
        int data;

        System.out.println("Presiona 0 para salir");
        System.out.println("Presiona 1 para un nuevo nodo");
        System.out.print("Ingresa la opción: ");

        // Manejo básico de entrada de enteros
        if (!scanner.hasNextInt()) {
            System.out.println("Entrada no válida. Terminando subárbol actual.");
            scanner.next(); // Limpia la entrada no válida
            return null;
        }
        choice = scanner.nextInt();

        if (choice == 0) {
            return null;
        } else if (choice == 1) {
            System.out.print("Ingresa los datos: ");
            data = scanner.nextInt();

            Node temp = new Node(data);

            System.out.println("Ingresa el nodo hijo izquierdo de " + data);
            temp.left = createTree(); // Llamada recursiva

            System.out.println("Ingresa el nodo hijo derecho de " + data);
            temp.right = createTree(); // Llamada recursiva

            return temp;
        } else {
            System.out.println("Opción no válida. Terminando subárbol actual.");
            return null;
        }
    }
}

/*
// Ejemplo de uso en la clase principal
public class Main {
    public static void main(String[] args) {
        Node root = BinaryTreeCreator.createTree();
        // Nota: El scanner de System.in permanece abierto en el ejemplo.
    }
}
*/

File: Árbol Binario Interactivo/Arbol.cs
using System;

// Clase para representar un nodo
public class Node
{
    public int data;
    public Node left;
    public Node right;

    public Node(int data)
    {
        this.data = data;
        this.left = null;
        this.right = null;
    }
}

public class BinaryTreeCreator
{
    // Función para crear un árbol binario de forma recursiva
    public static Node CreateTree()
    {
        int choice;
        int data;

        while (true)
        {
            Console.WriteLine("Presiona 0 para salir");
            Console.WriteLine("Presiona 1 para un nuevo nodo");
            Console.Write("Ingresa la opción: ");

            // Captura de entrada
            if (!int.TryParse(Console.ReadLine(), out choice))
            {
                Console.WriteLine("Entrada no válida. Intenta de nuevo.");
                continue;
            }

            if (choice == 0)
            {
                return null;
            }
            else if (choice == 1)
            {
                Console.Write("Ingresa los datos: ");
                if (!int.TryParse(Console.ReadLine(), out data))
                {
                    Console.WriteLine("Entrada de datos no válida. Intenta de nuevo.");
                    continue;
                }

                Node temp = new Node(data);

                Console.WriteLine($"Ingresa el nodo hijo izquierdo de {data}");
                temp.left = CreateTree(); // Llamada recursiva

                Console.WriteLine($"Ingresa el nodo hijo derecho de {data}");
                temp.right = CreateTree(); // Llamada recursiva

                return temp;
            }
            else
            {
                Console.WriteLine("Opción no válida. Intenta de nuevo.");
            }
        }
    }
}

/*
// Ejemplo de uso
public class Program
{
    public static void Main(string[] args)
    {
        Node root = BinaryTreeCreator.CreateTree();
        // Uso del árbol...
    }
}
*/